### PR TITLE
Bugfix start screen double mouse click register

### PR DESCRIPTION
**Pull request by Pathemeous**
Fixes the registering of multiple mousepresses when switching menus, causing users to fall through the menus (when buttons in different menus are in the same location).

The tests have been re-mocked, because the inputs are handled in different places now. This was necessary, because the isMousePressed can only be called once per button per update cycle, due to how its buffer works. Using isMouseButtonDown is what caused the bug in the first place, and evidently is not a solution for locally checking input.

While this causes an extra if statement, this cannot be helped due to how the input engine works.

< testing, bug

### CODE REVIEW

**Comment by BCrlrd** [comment]
Could you merge the dev branch into this branch so we can look at it without morge conflicts?

< review

**Comment by Pathemeous** [comment]
Done, one import line was conflicting.

< review, layout

File: src/main/java/nl/tudelft/controller/StartScreenStateController.java [hunk 1]
```diff
@@ -191,9 +190,9 @@ public MouseOverArea createQuitButton(GameContainer container) {
                 364, 88);
     }
 
-    public boolean isAreaClicked(MouseOverArea buttonArea, Input input) {
-        return input.isMouseButtonDown(Input.MOUSE_LEFT_BUTTON) && buttonArea.isMouseOver();
-    }
+//    public boolean isAreaClicked(MouseOverArea buttonArea, Input input) {
+//        return input.isMouseButtonDown(Input.MOUSE_LEFT_BUTTON) && buttonArea.isMouseOver();
```
Comment (BCrlrd) [inline src/main/java/nl/tudelft/controller/StartScreenStateController.java]: Commented method can be removed.

File: src/test/java/nl/tudelft/view/StartScreenStateTest.java [hunk 2]
```diff
@@ -207,8 +240,9 @@ public void testUpdatePlayer2ButtonClick() throws SlickException {
     // input.clearMousePressedRecord();
     // This test can be uncommented after these lines have been removed.
     // @Test
-    // public void testUpdateOptionsButtonClick() throws SlickException {
-    // Mockito.when(mockedController.isAreaClicked(Mockito.any(), Mockito.any())).thenReturn(
+    // public void testUpdateOptionsButtonHover() throws SlickException {
+    // state.init(mockedContainer, mockedSBG);
+    // Mockito.when(mockedMouseOverArea.isMouseOver()).thenReturn(
```
Comment (BCrlrd) [inline src/test/java/nl/tudelft/view/StartScreenStateTest.java]: Commented test should be deleted or fixed.

File: src/test/java/nl/tudelft/view/StartScreenStateTest.java [hunk 3]
```diff
@@ -46,17 +49,33 @@ public void setUp() {
...
+                mockedMouseOverArea);
+        Mockito.when(mockedController.createOptionsButton(Mockito.any())).thenReturn(
+                mockedMouseOverArea);
+        Mockito.when(mockedController.createHighscoresButton(Mockito.any())).thenReturn(
+                mockedMouseOverArea);
+        Mockito.when(mockedController.createQuitButton(Mockito.any())).thenReturn(
+                mockedMouseOverArea);
+
         mockedFont = Mockito.mock(TrueTypeFont.class);
         Mockito.when(mockedResources.createFont(new Font("Verdana", Font.BOLD, 36), true))
```
Comment (BCrlrd) [inline src/test/java/nl/tudelft/view/StartScreenStateTest.java]: new Font("Verdana", Font.BOLD, 36), true)
can be replaced by "any()" to improve readability, as you need to return a mocked font for every font which will/can be created. If you want to be certain you only return the mocked font when this particular font is created you should keep it as is.

**Comment by Pathemeous** [comment]
I implemented all your feedback.

< review

**Comment by BCrlrd** [comment]
After these changes I have nothing more to complain about. Will merge.

< review

**Pull request merged**